### ISSUE TITLE
PHP 5.6: Error because "$HTTP_RAW_POST_DATA"

### ISSUE THREAD

**Issue by bax-** [fmbiete/Z-Push-contrib#135]
When installing this on a standard PHP 5.6 install, the following error is being thrown and breaks the whole app:

```
[error] 27733#0: *14873 FastCGI sent in stderr: "PHP message: PHP Deprecated:  Automatically populating $HTTP_RAW_POST_DATA is deprecated and will be removed in a future version. To avoid this warning set 'always_populate_raw_post_data' to '-1' in php.ini and use the php://input stream instead. in Unknown on line 0" while reading response header from upstream, client: […]
```

The line in question is commented out in the default php.ini for 5.6. So you have to un-comment it.
Maybe there is something that could / should be changed in z-push for future releases' better compatibility.

http://php.net/manual/en/reserved.variables.httprawpostdata.php

https://www.bram.us/2014/10/26/php-5-6-automatically-populating-http_raw_post_data-is-deprecated-and-will-be-removed-in-a-future-version/

**Comment by fmbiete** [comment]
Maybe I'm missing something but Z-Push is not using HTTP_RAW_POST_DATA, we are already using the php://input file.
Or at least I don't find HTTP_RAW_POST_DATA anywhere in the code...

**Comment by bax-** [comment]
That's what I thought, too. I got the message after the upgrade anyways.
So it's probably more a php thing (the standard releases shouldn't have `always_populate_raw_post_data` commented out.
But I wasn't sure if there wasn't something in z-push still using a part of the HTTP_RAW_POST_DATA so I thought it was worth sharing, also for people upgrading in the future…

**Comment by fmbiete** [comment]
Yup, maybe it's a dependant library that includes the depecrated bit. I have just searched in awl library (xmldocument.php for caldav and carddav support), but it doesn't include it.
So maybe curl or some other system package.

I intend to upgrade my servers to 5.6 soon, so I will take a deeper look into this then.
